src/plugins/discover/public/application/components/source_viewer/source_viewer.tsx

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import React, { useEffect, useState } from 'react';
+import { monaco } from '@kbn/monaco';
+import { EuiEmptyPrompt, EuiLoadingSpinner } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { useEsDocSearch } from '../doc/use_es_doc_search';
+import { JsonCodeEditorCommon } from '../json_code_editor/json_code_editor_common';
+import { ElasticRequestState } from '../doc/elastic_request_state';
+import { getServices } from '../../../../public/kibana_services';
+import { SEARCH_FIELDS_FROM_SOURCE } from '../../../../common';
+
+interface SourceViewerProps {
+  id: string;
+  index: string;
+  indexPatternId: string;
+  hasLineNumbers: boolean;
+  width?: number;
+}
+
+export const SourceViewer = ({
+  id,
+  index,
+  indexPatternId,
+  width,
+  hasLineNumbers,
+}: SourceViewerProps) => {
+  const [editor, setEditor] = useState<monaco.editor.IStandaloneCodeEditor>();
+  const [jsonValue, setJsonValue] = useState<string>('');
+  const indexPatternService = getServices().data.indexPatterns;
+  const useNewFieldsApi = getServices().uiSettings.get(SEARCH_FIELDS_FROM_SOURCE);
+  const [reqState, hit] = useEsDocSearch({
+    id,
+    index,
+    indexPatternId,
+    indexPatternService,
+    requestAllFields: useNewFieldsApi,
+  });
+
+  useEffect(() => {
+    if (reqState === ElasticRequestState.Found) {
+      setJsonValue(JSON.stringify(hit, undefined, 2));
+    }
```

Comment:
this initial value of `hit` is `null`, so in this case it should `setJsonValue` should not be called